src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private static final String INDEX_OUT_OF_BOUNDS_EXCEPTION_MESSAGE
+            = "Invalid index value! Actual list size is: ";
+    private static final String NO_SUCH_ELEMENT_EXCEPTION_MESSAGE
+            = "Element not found! Invalid value!";
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = createListContainer(INITIAL_CAPACITY);
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
```

Comment:
Don't need create method shiftRight if you use it only in one place
```suggestion
        indexValidation(index);
        grow();
        System.arraycopy(elementData, index, elementData, index + 1, size - index);
        elementData[index] = value;
        size++;
```